src/renderer/App.tsx

```diff
@@ -99,21 +105,28 @@ class App extends Component<{}, State> {
         newState.options.selectAll = false;
       }
     }
-
     this.setState(newState);
   };
 
   selectNetwork: SelectNetwork = (network) => {
     this.setState({ view: 'networks', selectedNetwork: network });
   };
 
-  convertAndStoreYamlJSON = (yamlText: string) => {
+  convertAndStoreYamlJSON = (yamlText: string, filePath: string) => {
+    // Make copy of current state;
     const yamlJSON = yaml.safeLoad(yamlText);
-    const yamlState = convertYamlToState(yamlJSON);
+    const yamlState = convertYamlToState(yamlJSON, filePath);
+    const openFiles = this.state.openFiles.slice();
+    // Don't add a file that is already opened to the openFiles array
+    if (!openFiles.includes(filePath)) openFiles.push(filePath);
+
     // set global variables for d3 simulation
     window.d3State = setD3State(yamlState.services);
+
+    // store opened file state in localStorage under the current state item call "state" as well as an individual item using the filePath as the key.
     localStorage.setItem('state', JSON.stringify(yamlState));
-    this.setState(Object.assign(initialState, yamlState));
+    localStorage.setItem(`${filePath}`, JSON.stringify(yamlState));
```

Comment:
why are there two keys in local storage with the same state?

Reply:
This was not meant to be merged to this version of the code yet, we are still working. But 'state' key represents data of the current active file, whereas that data is also stored in a key specific to that file path. This is because the original team made it so that if a 'state' key existed in local storage when the app loads, it would pull that data and populate the view with it. So we decided to keep that 'state' key as not to break it. But then when switching tabs, that data held in the 'state' key would change to the next file. If we wanted to return to the first file, we had to store it under its own file path so it wouldn't get overwritten. It works well as we currently have it. The code you are looking at is from weeks ago.